src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -532,5 +559,53 @@ public String getContentType() {
     public String getOutputFileNameBase() {
       return outputFileNameBase;
     }
+
+    public GCSBatchSinkConfig() {
+      super();
+    }
+
+    //This method validated the pattern of CMEK Key resource ID.
+    void validateCmekKey(FailureCollector failureCollector) {
+      CryptoKeyName cmekKeyName = null;
+      try {
+        cmekKeyName = CryptoKeyName.parse(cmekKey);
+      } catch (ValidationException e) {
+        failureCollector.addFailure(e.getMessage(), null)
+          .withConfigProperty(NAME_CMEK_KEY).withStacktrace(e.getStackTrace());
+        return;
+      }
+      Boolean isServiceAccountFilePath = isServiceAccountFilePath();
+      Credentials credentials = null;
+      try {
+        credentials = getServiceAccount() == null ?
+          null : GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath);
+      } catch (Exception e) {
+        /*Ignoring the exception because we don't want to highlight cmek key if an exception occurs while
+        loading credentials*/
+        return;
+      }
+      Storage storage = GCPUtils.getStorage(getProject(), credentials);
+      Bucket bucket = null;
+      try {
+        bucket = storage.get(getBucket());
+      } catch (StorageException e) {
+        /* Ignoring the exception because we don't want the validation to fail if there is an exception getting
+          the bucket information either because the service account used during validation can be different than
+          the service account that will be used at runtime (the dataproc service account)
+          (assuming the user has auto-detect for the service account) */
+        return;
+      }
+      if (bucket == null) {
+        String cmekKeyLocation = cmekKeyName.getLocation();
+        if ((Strings.isNullOrEmpty(location) && "US".equalsIgnoreCase(cmekKeyLocation))
```

Comment:
shouldn't this be `!"US".equals...`? Please add a unit test for this scenario.

Reply:
Yeah, missed it will make the change